pkg/agent/client.go

```diff
@@ -479,25 +486,35 @@ func (a *Client) remoteToProxy(connID int64, ctx *connContext) {
 		klog.V(5).InfoS("received data from remote", "bytes", n, "connectionID", connID)
 
 		if err == io.EOF {
-			klog.V(2).Infoln("connection EOF")
+			klog.V(2).InfoS("connection EOF", "connectionID", connID)
 			return
 		} else if err != nil {
 			// Normal when receive a CLOSE_REQ
-			klog.ErrorS(err, "connection read failure")
+			klog.ErrorS(err, "connection read failure", "connectionID", connID)
 			return
 		} else {
 			resp.Payload = &client.Packet_Data{Data: &client.Data{
 				Data:      buf[:n],
 				ConnectID: connID,
 			}}
 			if err := a.Send(resp); err != nil {
-				klog.ErrorS(err, "stream send failure")
+				klog.ErrorS(err, "stream send failure", "connectionID", connID)
 			}
 		}
 	}
 }
 
 func (a *Client) proxyToRemote(connID int64, ctx *connContext) {
+	defer func() {
+		if err := recover(); err != nil {
+			klog.V(2).InfoS("Exiting proxyToRemote with recovery", "error", err, "connectionID", connID)
+		} else {
+			klog.V(2).InfoS("Exiting proxyToRemote", "connectionID", connID)
+		}
+	}()
+	defer func() {
```

Comment:
Remove this defer? Because we covered this in the above defer.
return of recover() is nil only when no panic happens, so L512 works in the same way.

Reply:
Sorry this was leftover which should have been removed.